File: vivo_blueos_input/01-Primary/no_comments/test-queue.c
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>
#include "alloc-testing.h"
#include "framework.h"
#include "queue.h"
int variable1, variable2, variable3, variable4;
Queue *generate_queue(void)
{
	Queue *queue;
	int i;
	queue = queue_new();
	for (i=0; i<1000; ++i) {
		queue_push_head(queue, &variable1);
		queue_push_head(queue, &variable2);
		queue_push_head(queue, &variable3);
		queue_push_head(queue, &variable4);
	}
	return queue;
}
void test_queue_new_free(void)
{
	int i;
	Queue *queue;
	queue = queue_new();
	queue_free(queue);
	queue = queue_new();
	for (i=0; i<1000; ++i) {
		queue_push_head(queue, &variable1);
	}
	queue_free(queue);
	alloc_test_set_limit(0);
	queue = queue_new();
	assert(queue == NULL);
}
void test_queue_push_head(void)
{
	Queue *queue;
	int i;
	queue = queue_new();
	for (i=0; i<1000; ++i) {
		queue_push_head(queue, &variable1);
		queue_push_head(queue, &variable2);
		queue_push_head(queue, &variable3);
		queue_push_head(queue, &variable4);
	}
	assert(!queue_is_empty(queue));
	assert(queue_pop_tail(queue) == &variable1);
	assert(queue_pop_tail(queue) == &variable2);
	assert(queue_pop_tail(queue) == &variable3);
	assert(queue_pop_tail(queue) == &variable4);
	assert(queue_pop_head(queue) == &variable4);
	assert(queue_pop_head(queue) == &variable3);
	assert(queue_pop_head(queue) == &variable2);
	assert(queue_pop_head(queue) == &variable1);
	queue_free(queue);
	queue = queue_new();
	alloc_test_set_limit(0);
	assert(!queue_push_head(queue, &variable1));
	queue_free(queue);
}
void test_queue_pop_head(void)
{
	Queue *queue;
	queue = queue_new();
	assert(queue_pop_head(queue) == NULL);
	queue_free(queue);
	queue = generate_queue();
	while (!queue_is_empty(queue)) {
		assert(queue_pop_head(queue) == &variable4);
		assert(queue_pop_head(queue) == &variable3);
		assert(queue_pop_head(queue) == &variable2);
		assert(queue_pop_head(queue) == &variable1);
	}
	assert(queue_pop_head(queue) == NULL);
	queue_free(queue);
}
void test_queue_peek_head(void)
{
	Queue *queue;
	queue = queue_new();
	assert(queue_peek_head(queue) == NULL);
	queue_free(queue);
	queue = generate_queue();
	while (!queue_is_empty(queue)) {
		assert(queue_peek_head(queue) == &variable4);
		assert(queue_pop_head(queue) == &variable4);
		assert(queue_peek_head(queue) == &variable3);
		assert(queue_pop_head(queue) == &variable3);
		assert(queue_peek_head(queue) == &variable2);
		assert(queue_pop_head(queue) == &variable2);
		assert(queue_peek_head(queue) == &variable1);
		assert(queue_pop_head(queue) == &variable1);
	}
	assert(queue_peek_head(queue) == NULL);
	queue_free(queue);
}
void test_queue_push_tail(void)
{
	Queue *queue;
	int i;
	queue = queue_new();
	for (i=0; i<1000; ++i) {
		queue_push_tail(queue, &variable1);
		queue_push_tail(queue, &variable2);
		queue_push_tail(queue, &variable3);
		queue_push_tail(queue, &variable4);
	}
	assert(!queue_is_empty(queue));
	assert(queue_pop_head(queue) == &variable1);
	assert(queue_pop_head(queue) == &variable2);
	assert(queue_pop_head(queue) == &variable3);
	assert(queue_pop_head(queue) == &variable4);
	assert(queue_pop_tail(queue) == &variable4);
	assert(queue_pop_tail(queue) == &variable3);
	assert(queue_pop_tail(queue) == &variable2);
	assert(queue_pop_tail(queue) == &variable1);
	queue_free(queue);
	queue = queue_new();
	alloc_test_set_limit(0);
	assert(!queue_push_tail(queue, &variable1));
	queue_free(queue);
}
void test_queue_pop_tail(void)
{
	Queue *queue;
	queue = queue_new();
	assert(queue_pop_tail(queue) == NULL);
	queue_free(queue);
	queue = generate_queue();
	while (!queue_is_empty(queue)) {
		assert(queue_pop_tail(queue) == &variable1);
		assert(queue_pop_tail(queue) == &variable2);
		assert(queue_pop_tail(queue) == &variable3);
		assert(queue_pop_tail(queue) == &variable4);
	}
	assert(queue_pop_tail(queue) == NULL);
	queue_free(queue);
}
void test_queue_peek_tail(void)
{
	Queue *queue;
	queue = queue_new();
	assert(queue_peek_tail(queue) == NULL);
	queue_free(queue);
	queue = generate_queue();
	while (!queue_is_empty(queue)) {
		assert(queue_peek_tail(queue) == &variable1);
		assert(queue_pop_tail(queue) == &variable1);
		assert(queue_peek_tail(queue) == &variable2);
		assert(queue_pop_tail(queue) == &variable2);
		assert(queue_peek_tail(queue) == &variable3);
		assert(queue_pop_tail(queue) == &variable3);
		assert(queue_peek_tail(queue) == &variable4);
		assert(queue_pop_tail(queue) == &variable4);
	}
	assert(queue_peek_tail(queue) == NULL);
	queue_free(queue);
}
void test_queue_is_empty(void)
{
	Queue *queue;
	queue = queue_new();
	assert(queue_is_empty(queue));
	queue_push_head(queue, &variable1);
	assert(!queue_is_empty(queue));
	queue_pop_head(queue);
	assert(queue_is_empty(queue));
	queue_push_tail(queue, &variable1);
	assert(!queue_is_empty(queue));
	queue_pop_tail(queue);
	assert(queue_is_empty(queue));
	queue_free(queue);
}
static UnitTestFunction tests[] = {
	test_queue_new_free,
	test_queue_push_head,
	test_queue_pop_head,
	test_queue_peek_head,
	test_queue_push_tail,
	test_queue_pop_tail,
	test_queue_peek_tail,
	test_queue_is_empty,
	NULL
};
int main(int argc, char *argv[])
{
	run_tests(tests);
	return 0;
}

File: Output/primary/src/test_queue.rs
use crate::queue::*;
use crate::translation_utils::*;static mut variable1: i32 = 0;
static mut variable2: i32 = 0;
static mut variable3: i32 = 0;
static mut variable4: i32 = 0;
fn generate_queue() -> Manual<Queue<Ptr<i32>>> {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>> = queue_new();
        let mut i: i32;
        c_for!(i = 0; i < 1000; i += 1; {
            queue_push_head(queue, c_ref!(variable1));
            queue_push_head(queue, c_ref!(variable2));
            queue_push_head(queue, c_ref!(variable3));
            queue_push_head(queue, c_ref!(variable4));
        });
        return queue;
    }
}
#[test]
fn test_queue_new_free() {
    unsafe {
        let mut i: i32;
        let mut queue: Manual<Queue<Ptr<i32>>>;
        queue = queue_new();
        queue_free(queue);
        queue = queue_new();
        c_for!(i = 0; i < 1000; i += 1; {
            queue_push_head(queue, c_ref!(variable1));
        });
        queue_free(queue);
        test_no_memory_leak!();
    }
}
#[test]
fn test_queue_push_head() {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>>;
        let mut i: i32;
        queue = queue_new();
        c_for!(i = 0; i < 1000; i += 1; {
            queue_push_head(queue, c_ref!(variable1));
            queue_push_head(queue, c_ref!(variable2));
            queue_push_head(queue, c_ref!(variable3));
            queue_push_head(queue, c_ref!(variable4));
        });
        assert!(!queue_is_empty(queue).as_bool());
        assert!(queue_pop_tail(queue) == c_ref!(variable1));
        assert!(queue_pop_tail(queue) == c_ref!(variable2));
        assert!(queue_pop_tail(queue) == c_ref!(variable3));
        assert!(queue_pop_tail(queue) == c_ref!(variable4));
        assert!(queue_pop_head(queue) == c_ref!(variable4));
        assert!(queue_pop_head(queue) == c_ref!(variable3));
        assert!(queue_pop_head(queue) == c_ref!(variable2));
        assert!(queue_pop_head(queue) == c_ref!(variable1));
        queue_free(queue);
        test_no_memory_leak!();
    }
}
#[test]
fn test_queue_pop_head() {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>>;
        queue = queue_new();
        assert!(queue_pop_head(queue) == null!());
        queue_free(queue);
        queue = generate_queue();
        while !queue_is_empty(queue).as_bool() {
            assert!(queue_pop_head(queue) == c_ref!(variable4));
            assert!(queue_pop_head(queue) == c_ref!(variable3));
            assert!(queue_pop_head(queue) == c_ref!(variable2));
            assert!(queue_pop_head(queue) == c_ref!(variable1));
        }
        assert!(queue_pop_head(queue) == null!());
        queue_free(queue);
        test_no_memory_leak!();
    }
}
#[test]
fn test_queue_peek_head() {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>>;
        queue = queue_new();
        assert!(queue_peek_head(queue) == null!());
        queue_free(queue);
        queue = generate_queue();
        while !queue_is_empty(queue).as_bool() {
            assert!(queue_peek_head(queue) == c_ref!(variable4));
            assert!(queue_pop_head(queue) == c_ref!(variable4));
            assert!(queue_peek_head(queue) == c_ref!(variable3));
            assert!(queue_pop_head(queue) == c_ref!(variable3));
            assert!(queue_peek_head(queue) == c_ref!(variable2));
            assert!(queue_pop_head(queue) == c_ref!(variable2));
            assert!(queue_peek_head(queue) == c_ref!(variable1));
            assert!(queue_pop_head(queue) == c_ref!(variable1));
        }
        assert!(queue_peek_head(queue) == null!());
        queue_free(queue);
        test_no_memory_leak!();
    }
}
#[test]
fn test_queue_push_tail() {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>>;
        let mut i: i32;
        queue = queue_new();
        c_for!(i = 0; i < 1000; i += 1; {
            queue_push_tail(queue, c_ref!(variable1));
            queue_push_tail(queue, c_ref!(variable2));
            queue_push_tail(queue, c_ref!(variable3));
            queue_push_tail(queue, c_ref!(variable4));
        });
        assert!(!queue_is_empty(queue).as_bool());
        assert!(queue_pop_head(queue) == c_ref!(variable1));
        assert!(queue_pop_head(queue) == c_ref!(variable2));
        assert!(queue_pop_head(queue) == c_ref!(variable3));
        assert!(queue_pop_head(queue) == c_ref!(variable4));
        assert!(queue_pop_tail(queue) == c_ref!(variable4));
        assert!(queue_pop_tail(queue) == c_ref!(variable3));
        assert!(queue_pop_tail(queue) == c_ref!(variable2));
        assert!(queue_pop_tail(queue) == c_ref!(variable1));
        queue_free(queue);
        test_no_memory_leak!();
    }
}
#[test]
fn test_queue_pop_tail() {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>>;
        queue = queue_new();
        assert!(queue_pop_tail(queue) == null!());
        queue_free(queue);
        queue = generate_queue();
        while !queue_is_empty(queue).as_bool() {
            assert!(queue_pop_tail(queue) == c_ref!(variable1));
            assert!(queue_pop_tail(queue) == c_ref!(variable2));
            assert!(queue_pop_tail(queue) == c_ref!(variable3));
            assert!(queue_pop_tail(queue) == c_ref!(variable4));
        }
        assert!(queue_pop_tail(queue) == null!());
        queue_free(queue);
        test_no_memory_leak!();
    }
}
#[test]
fn test_queue_peek_tail() {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>>;
        queue = queue_new();
        assert!(queue_peek_tail(queue) == null!());
        queue_free(queue);
        queue = generate_queue();
        while !queue_is_empty(queue).as_bool() {
            assert!(queue_peek_tail(queue) == c_ref!(variable1));
            assert!(queue_pop_tail(queue) == c_ref!(variable1));
            assert!(queue_peek_tail(queue) == c_ref!(variable2));
            assert!(queue_pop_tail(queue) == c_ref!(variable2));
            assert!(queue_peek_tail(queue) == c_ref!(variable3));
            assert!(queue_pop_tail(queue) == c_ref!(variable3));
            assert!(queue_peek_tail(queue) == c_ref!(variable4));
            assert!(queue_pop_tail(queue) == c_ref!(variable4));
        }
        assert!(queue_peek_tail(queue) == null!());
        queue_free(queue);
        test_no_memory_leak!();
    }
}
#[test]
fn test_queue_is_empty() {
    unsafe {
        let mut queue: Manual<Queue<Ptr<i32>>>;
        queue = queue_new();
        assert!(queue_is_empty(queue).as_bool());
        queue_push_head(queue, c_ref!(variable1));
        assert!(!queue_is_empty(queue).as_bool());
        queue_pop_head(queue);
        assert!(queue_is_empty(queue).as_bool());
        queue_push_tail(queue, c_ref!(variable1));
        assert!(!queue_is_empty(queue).as_bool());
        queue_pop_tail(queue);
        assert!(queue_is_empty(queue).as_bool());
        queue_free(queue);
        test_no_memory_leak!();
    }
}
